sdk/spring/spring-cloud-azure-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/aad/properties/AadAuthenticationProperties.java

```diff
@@ -552,100 +568,115 @@ private void validateApplicationType() {
-            if (!scopes.contains("profile")) {
-                scopes.add("profile"); // "profile" allows returning additional claims in the ID token.
-            }
-            if (!scopes.contains("offline_access")) {
-                scopes.add("offline_access"); // "offline_access" allows to request a refresh token.
+        return scopes;
+    }
+
+    private void validateAuthorizationGrantType(String registrationId, AadAuthorizationGrantType grantType,
+                                                AuthorizationClientProperties properties) {
+
+        if (compatibleApplicationTypeAndGrantTypes.containsKey(applicationType)) {
+            if (compatibleApplicationTypeAndGrantTypes.get(applicationType).contains(grantType)) {
+                throw new IllegalStateException(String.format(UNMATCHING_OAUTH_GRANT_TYPE_FROMAT,
+                    applicationType.getValue(), registrationId, grantType));
             }
+            LOGGER.debug("'spring.cloud.azure.active-directory.authorization-clients.{}.authorization-grant-type'"
+                + " is valid.", registrationId);
+        }
+
+        if (AZURE_CLIENT_REGISTRATION_ID.equals(registrationId)
+            && AUTHORIZATION_CODE != properties.getAuthorizationGrantType()) {
+            throw new IllegalStateException("spring.cloud.azure.active-directory.authorization-clients."
+                + AZURE_CLIENT_REGISTRATION_ID
+                + ".authorization-grant-type must be configured to 'authorization_code'.");
         }
     }
 
+    /**
+     * Decide the default grant type from application type.
+     * @param registrationId client ID
+     * @param appType AadApplicationType
+     * @return default grant type
+     */
+    private AadAuthorizationGrantType decideDefaultGrantTypeFromApplicationType(String registrationId,
+                                                                                AadApplicationType appType) {
+        AadAuthorizationGrantType grantType;
+        switch (appType) {
+            case WEB_APPLICATION:
+                if (registrationId.equals(AZURE_CLIENT_REGISTRATION_ID)) {
+                    grantType = AUTHORIZATION_CODE;
+                } else {
+                    grantType = AZURE_DELEGATED;
+                }
+                break;
+            case RESOURCE_SERVER:
+            case RESOURCE_SERVER_WITH_OBO:
+                grantType = AadAuthorizationGrantType.ON_BEHALF_OF;
+                break;
+            case WEB_APPLICATION_AND_RESOURCE_SERVER:
+                throw new IllegalStateException("spring.cloud.azure.active-directory.authorization-clients." + registrationId
+                    + ".authorization-grant-grantType must be configured. ");
+            default:
+                throw new IllegalStateException("Unsupported authorization grantType " + appType.getValue());
+        }
+
+        LOGGER.debug("The client '{}' sets the default value of AADAuthorizationGrantType to '{}'.", grantType,
```

Comment:
we should change the log info or move the log to where the grantType is set to the property

Reply:
OK